comps/llms/text-generation/predictionguard/predictionguard_llm.py

```diff
@@ -1,64 +1,57 @@
 # Copyright (C) 2024 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
+# TODO: Add streaming
 import os
 from fastapi.responses import StreamingResponse
-from langsmith import traceable
-from comps import GeneratedDoc, LLMParamsDoc, ServiceType, opea_microservices, register_microservice
+from fastapi import FastAPI, HTTPException
+from pydantic import BaseModel
 from predictionguard import PredictionGuard
-import os
-import requests
-from langsmith import traceable
+from comps import GeneratedDoc, ServiceType, opea_microservices, register_microservice
+
+# Set up Prediction Guard API key
+PG_API_KEY = os.getenv("PREDICTIONGUARD_API_KEY")
+
+# Define the input data model
+class LLMParamsDoc(BaseModel):
+    model: str = "Neural-Chat-7B"
+    query: str
+    max_new_tokens: int
+    temperature: float
+    top_p: float
+    top_k: int
+    #stream: bool = False
 
-# Initialize Prediction Guard API URL and Key
-PG_API_URL = "https://api.predictionguard.com/completions"
-PG_API_KEY = os.getenv("PREDICTION_GUARD_API_KEY")
+app = FastAPI()
 
 @register_microservice(
     name="opea_service@llm_predictionguard",
     service_type=ServiceType.LLM,
-    endpoint="/v1/chat/completions",
+    endpoint="/v1/completions",
```

Comment:
Is this replacing or adding a new endpoint? Does this branch need to be updated with the latest from main?

Reply:
I am not sure I understand correctly. But this is an example of how to run it:
```
curl -X POST http://localhost:9000/v1/completions \
     -H "Content-Type: application/json" \
     -d '{
         "model": "Neural-Chat-7B",
         "query": "Tell me a joke.",
         "max_new_tokens": 100,
         "temperature": 0.7,
         "top_p": 0.9,
         "top_k": 50
     }
    ```
    
    Let me know if I need to make any changes

Reply:
Nvm, I got confused by the changes. I assume we won't be exposing a `/chat/completions` endpoint then?

Reply:
I will replace with the chat/completions end point